mmcv/ops/csrc/common/cuda/convex_iou_cuda_kernel.cuh

```diff
@@ -18,9 +18,37 @@ struct Point {
   double x, y;
   __device__ Point() {}
   __device__ Point(double x, double y) : x(x), y(y) {}
+  __device__ friend bool operator<(const Point& lhs, const Point& rhs) {
+    if(lhs.y != rhs.y) return lhs.y < rhs.y;
+    if(lhs.x != rhs.x) return lhs.x < rhs.x;
+    return true;
+  }
 };
 
-__device__ inline bool point_same(Point& a, Point& b) {
+__device__ inline void quickSort(Point *ps, int elements) {
```

Comment:
Have you profile the kernel with different sort algrithm? kernels with large register usage might have pool occupancy on gpu.
https://docs.nvidia.com/gameworks/content/developertools/desktop/analysis/report/cudaexperiments/kernellevel/achievedoccupancy.htm

Reply:
You can use thrust inside device function to perform the sort.

Reply:
Does it mean the parallel algorithms library --- [thrust](https://thrust.github.io/)?

Reply:
Yes, cuda has its built-in thrust. Or you can implement the sort by yourself with less register usage.

Reply:
I compare the quickSort with thrust::sort in terms of the time overhead in the following table, which presents the average time overhead. 
| blocks   | thrust::sort         | quickSort             |
|----------|----------------------|-----------------------|
| 65536    | 0.007929563522338867 | 0.0036458969116210938 |
| 65536*8  | 0.01790785789489746  | 0.013720512390136719  |
| 65536*16 | 0.031015396118164062 | 0.027451753616333008  |
| 65536*32 | 0.05516695976257324  | 0.05521392822265625   |

Reply:
Sure, that's cool. You can keep quickSort here.